### PR TITLE
CDAP-4535 Make announce address of Auth Server configurable

### PR DESCRIPTION
**Pull request by ecfm**
This PR is the same as PR https://github.com/caskdata/cdap/pull/6592

https://issues.cask.co/browse/CDAP-4535
http://builds.cask.co/browse/CDAP-DUT4752-2

### CODE REVIEW

**Comment by poornachandra** [comment]
Verified that this PR has the same commits as PR https://github.com/caskdata/cdap/pull/6592.

@maochf Please squash the commits. Once the build is green this is good to merge.

**Comment by ecfm** [comment]
Build is successful and the branch is ready to merge http://builds.cask.co/browse/CDAP-DUT4752-2

File: cdap-gateway/src/test/java/co/cask/cdap/gateway/router/AuthServerAnnounceTest.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+      router =
+        new NettyRouter(cConf, sConfiguration, InetAddresses.forString(hostname),
+                        new RouterServiceLookup((DiscoveryServiceClient) discoveryService,
+                                                new RouterPathLookup()),
+                        new MissingTokenValidator(), accessTokenTransformer, discoveryServiceClient);
+      router.startAndWait();
+
+      for (Map.Entry<Integer, String> entry : router.getServiceLookup().getServiceMap().entrySet()) {
+        serviceMap.put(entry.getValue(), entry.getKey());
```
Comment (maochf) [inline cdap-gateway/src/test/java/co/cask/cdap/gateway/router/AuthServerAnnounceTest.java]: `serviceMap.putAll(router.getServiceLookup().getServiceMap());`?

File: cdap-gateway/src/test/java/co/cask/cdap/gateway/router/AuthServerAnnounceTest.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+      Map<String, List<String>> responseMap =
+        GSON.fromJson(new InputStreamReader(response.getEntity().getContent()), TYPE);
+      Assert.assertEquals(Collections.EMPTY_LIST, responseMap.get("auth_uri"));
+    } finally {
+      routerService.shutDown();
+    }
+  }
+
+  @Test
+  public void testAnnounceAddressConfig() throws Exception {
```
Comment (maochf) [inline cdap-gateway/src/test/java/co/cask/cdap/gateway/router/AuthServerAnnounceTest.java]: create a helper function to reduce duplicated code in `testEmptyAnnounceAddressConfig ` `testAnnounceAddressConfig`?

**Pull request merged**